pkg/build/builder/cmd/scmauth/password.go

```
@@ -0,0 +1,107 @@
+
+	// Only apply to https and http URLs
+	if scheme := strings.ToLower(u.SourceURL.Scheme); scheme != "http" && scheme != "https" {
+		return nil
+	}
+
+	// Determine username
+	// 1. Look for a username secret
+	username, err := readSecret(baseDir, UsernameSecret)
+	if err != nil {
+		return err
+	}
+	// 2. If not provided, look at the username in the URL
+	if username == "" && u.SourceURL.User != nil {
+		username = u.SourceURL.User.Username()
+	}
+	// 3. If still not found, use a default username
+	if username == "" {
+		username = DefaultUsername
+	}
+
+	// Determine password
+	// 1. Look for a token secret
+	password, err := readSecret(baseDir, TokenSecret)
+	if err != nil {
+		return err
+	}
+	// 2. Look for a password secret
+	if password == "" {
+		password, err = readSecret(baseDir, PasswordSecret)
+		if err != nil {
+			return err
+		}
+	}
+	// 3. Look for a password in the URL
+	if password == "" && u.SourceURL.User != nil {
+		password, _ = u.SourceURL.User.Password()
+	}
+
+	gitconfig, err := ioutil.TempFile("", "gituserpass.")
+	if err != nil {
+		return err
+	}
+	defer gitconfig.Close()
+
+	usernamePasswordURL := u.SourceURL
+	usernamePasswordURL.User = url.UserPassword(username, password)
+	gitconfig.WriteString(fmt.Sprintf(UserPassGitConfig, usernamePasswordURL.String(), u.SourceURL.String()))
+
+	return ensureGitConfig(gitconfig.Name())
+}
+
+// Name returns the name of this auth method.
+func (_ UsernamePassword) Name() string {
+	return UsernamePasswordName
+}
+
+// Handles returns true if a username, password or token secret is present
+func (_ UsernamePassword) Handles(name string) bool {
+	switch name {
+	case UsernameSecret, PasswordSecret, TokenSecret:
+		return true
+	}
+	return false
+}
+
+// readSecret reads the contents of a secret file
+func readSecret(baseDir, fileName string) (string, error) {
+	file := filepath.Join(baseDir, fileName)
+	if _, err := os.Stat(file); err != nil {
+		return "", nil
```

swallowing the error is a little odd